doc/sf/number_series.qbk

```diff
@@ -259,9 +259,63 @@ Passing a value greater than `max_prime` results in a __domain_error being raise
 
 This function is `constexpr` only if the compiler supports C++14 constexpr functions.
 
-[endsect] [/section:primes]
+[endsect] [/section:primes Prime Numbers]
 
-[endsect] [/Number Series]
+[section:prime_sieve Prime Sieve]
+
+[h4 Synopsis]
+
+``
+#include <boost/math/special_functions/prime_sieve.hpp>
+``
+
+namespace boost { namespace math {
+
+    template<class ExecutionPolicy, typename Z, class OutputIterator>
+    auto prime_sieve(ExecutionPolicy&& policy, Z upper_bound, OutputIterator output) -> decltype(output)
+
+    template<class ExecutionPolicy, class Z, class OutputIterator>
+    auto prime_range(ExecutionPolicy&& policy, Z lower_bound, Z upper_bound, OutputIterator output) -> decltype(output)
+
+    template<typename Z, class OutputIterator>
+    auto prime_sieve(Z upper_bound, OutputIterator output) -> decltype(output)
+
+    template<class Z, class OutputIterator>
+    auto prime_range(Z lower_bound, Z upper_bound, OutputIterator output) -> decltype(output)
+
+
+}} // End namespaces
+
+[h4 Description]
+
+There are two sets of functions available `prime_sieve` and `prime_range`. `prime_sieve` will return all primes in the
+range [2, `upper_bound`). `prime_range` will return all the primes in the range [lower_bound, upper_bound).
+
+If you have a C++17 compatible compiler you are able to pass an execution policy to both functions. Any
+policy besides `std::execution::seq` will enable internal multi-threading. If your compiler is not C++17 compatible the
```

Comment:
Wait, is this correct? Shouldn't we throw an error if we get an esoteric execution policy? Which ones has compatible semantics with your implementation?

(I'll ping @jzmaddock on this one because my knowledge on this is so poor.)

Reply:
I fear Nick is correct, for example C++20 draft says:

> class execution::unsequenced_policy { unspecified };
> 1 The class unsequenced_policy is an execution policy type used as a unique type to disambiguate parallel
> algorithm overloading and indicate that a parallel algorithm’s execution may be vectorized, e.g., executed on
> a single thread using instructions that operate on multiple data items.

So that should be single threaded execution, since we don't actually support vectorisation.

I would be inclined to say:

* sequenced_policy, unsequenced_policy, single thread code.
* parallel_policy or parallel_unsequenced_policy, multi threaded code.
* Anything else.... well assuming anything else that's added follows the "may be" wording, I would say should be single threaded.  But they could equally be compiler errors on the grounds that we don't know what to do.

Reply:
@jzmaddock That makes sense to me. GCC seems to ignore vectorisation if it can not accomplish it, but Intel C++ throws an error telling you if it is not SIMD compatible. I do not make any internal calls to execution policies anymore so the ones passed in are purely used in if statements to determine whether to run the sequential functions or not.